server/src/main/java/com/epam/aidial/core/server/controller/DeploymentController.java

```diff
@@ -62,18 +65,39 @@ public Future<?> getDeployments() {
         return context.respond(HttpStatus.OK, list);
     }
 
-    public static Future<Deployment> selectDeployment(ProxyContext context, String id) {
+    public static Future<Deployment> selectDeployment(ProxyContext context, String id, boolean filterCustomProperties, boolean modifyEndpoint) {
         Deployment deployment = context.getConfig().selectDeployment(id);
-
+        Proxy proxy = context.getProxy();
         if (deployment != null) {
             if (!deployment.hasAccess(context.getUserRoles())) {
                 return Future.failedFuture(new PermissionDeniedException("Forbidden deployment: " + id));
             } else {
-                return Future.succeededFuture(deployment);
+                try {
+                    if (deployment instanceof Application application) {
+                        if (!modifyEndpoint && !filterCustomProperties) {
+                            return Future.succeededFuture(deployment);
+                        }
+                        return proxy.getVertx().executeBlocking(() -> {
```

Comment:
I didn't get why we need to run the code in a worker thread?
I don't see any blocking ops there